packages/onchainkit/src/identity/hooks/useAddresses.ts

```diff
@@ -0,0 +1,31 @@
+import { getAddresses } from '@/identity/utils/getAddresses';
+import { DEFAULT_QUERY_OPTIONS } from '@/internal/constants';
+import { useQuery } from '@tanstack/react-query';
+import { mainnet } from 'viem/chains';
+import type {
+  GetAddressReturnType,
+  UseAddressesOptions,
+  UseQueryOptions,
+} from '../types';
+
+/**
+ * A React hook that leverages the `@tanstack/react-query` for fetching and optionally caching
+ * multiple Ethereum addresses from ENS names or Basenames in a single batch request.
+ */
+export const useAddresses = (
+  { names, chain = mainnet }: UseAddressesOptions,
+  queryOptions?: UseQueryOptions<GetAddressReturnType[]>,
+) => {
+  const namesKey = names.join(',');
+  const queryKey = ['useAddresses', namesKey, chain.id];
+
+  return useQuery<GetAddressReturnType[]>({
+    queryKey,
```

Comment:
suggestion(non-blocking): 100% personal preference, but if you wanna save some variable declaraions...

```tsx
queryKey: ['useAddresses', ...names, chain.id],
```